k8s/vizier/bootstrap/updater_role.yaml

```diff
@@ -1,77 +0,0 @@
----
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: pl-updater-service-account
----
-apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRoleBinding
-metadata:
-  name: pl-updater-cluster-binding
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: ClusterRole
-  name: pl-updater-role
-subjects:
-- kind: ServiceAccount
-  name: pl-updater-service-account
-  namespace: pl
----
-apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRole
-metadata:
-  name: pl-updater-cluster-role
-rules:
-# Allow actions on Kubernetes objects
-- apiGroups:
-  - rbac.authorization.k8s.io
-  - etcd.database.coreos.com
-  - nats.io
-  resources:
-  - clusterroles
-  - clusterrolebindings
-  - persistentvolumes
-  - etcdclusters
-  - natsclusters
-  verbs: ["*"]
----
-apiVersion: rbac.authorization.k8s.io/v1
-kind: Role
-metadata:
-  name: pl-updater-role
-rules:
-- apiGroups:
-  - ""
-  - apps
-  - rbac.authorization.k8s.io
-  - extensions
-  - batch
-  - policy
-  resources:
-  - configmaps
-  - secrets
-  - pods
-  - services
-  - deployments
-  - daemonsets
-  - persistentvolumes
-  - roles
-  - rolebindings
-  - serviceaccounts
-  - statefulsets
-  - cronjobs
-  - jobs
-  verbs: ["*"]
----
-apiVersion: rbac.authorization.k8s.io/v1
-kind: RoleBinding
-metadata:
-  name: pl-updater-binding
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: Role
-  name: pl-updater-role
-subjects:
-- kind: ServiceAccount
-  name: pl-updater-service-account
-  namespace: pl
```

Comment:
I believe these are still used, see this inlined template: https://github.com/pixie-io/pixie/blob/main/src/vizier/services/cloud_connector/bridge/server.go#L69

This is used in non-operator deploys of Pixie where we create a update job to update pixie.

Reply:
Ok... but it is almost certgainly broken in its current state: https://github.com/pixie-io/pixie/issues/1353

Do we have any tests to cover that non-operator deploy of Pixie?

Should I leave it, even though it won't work?

Reply:
Other alternatives that I'm open to:

1. attempt to _fix_ the ClusterRole naming issue in 1353, but again I'd be looking for guidance on how to test that change.
2. remove only the broken ClusterRole and ClusterRoleBindings identified in #1353 , but leave the Role, RoleBinding and ServiceAccount.
3. Entirely remove the non-operator update functionality??

Let me know what would be best.

Reply:
I was suggesting option 2.
i.e. keep `pl-updater-binding`, `pl-updater-role`, `pl-updater-service-account` and remove the rest.

Your assessment that the ClusterRole and the ClusterRoleBindings are incorrect seems accurate and I believe that it's a leftover from some old refactoring to scope down these roles.

Reply:
Re: non-operator deploys, they are here to stay for now and we might invest more into providing those as an explicit deployment option for pixie users. For now, we test both operator and non-operator releases/updates as part of our pre-release process. :)